app/utils/parse-vulnerable-api-finding.ts

```diff
@@ -188,15 +234,15 @@ function processFirstLine(
   if (lines[0]) {
     const firstLine = lines[0].trim();
     const colonIndex = firstLine.indexOf(': ');
+    const beforeColon = firstLine.substring(0, colonIndex).trim();
 
-    const genericUrlRegex = /^http[s]?:\/\/[a-zA-Z0-9.-]+(:\d+)?\/[^\s:]+/;
+    const genericUrlRegex =
+      /^(https?:\/\/)?([a-zA-Z0-9-]+\.)+[a-zA-Z0-9.-]+(:\d+)?(\/[^\s]*)?$/;
 
-    const match = firstLine.match(genericUrlRegex);
+    const match = beforeColon.match(genericUrlRegex);
 
     if (match) {
-      finding.request.url = finding.request.url = firstLine
-        .substring(0, colonIndex)
-        .trim();
+      finding.url = firstLine.substring(0, colonIndex).trim();
     }
 
     if (!firstLine.startsWith('confidence:')) {
```

Comment:
Can we merge the two if statements? 

`if (!firstLine.startsWith('confidence:') && colonIndex !== -1)`